distribution/lib/Standard/Base/0.0.0-dev/src/Polyglot/Proxy_Polyglot_Array.enso

```diff
@@ -0,0 +1,27 @@
+from Standard.Base import Polyglot, Array
+
+## Advanced
+
+    Wrapper for Polyglot Arrays
+type Proxy_Polyglot_Array
+
+    type Proxy_Polyglot_Array arr
+
+    ## Returns the number of elements stored in this Polyglot Array.
+
+    length : Number
+    length self =
+        Polyglot.get_array_size self.arr
+
+    ## Gets an element from this Polyglot Array at a specified index (0-based).
+
+    at : Number -> Any
+    at self index =
+        Polyglot.read_array_element self.arr index
+
+    to_array : Array Any
+    to_array self =
+        len = self.length
+        a = Array.new len
+        0.up_to len . each ix-> a.set_at ix (self.at ix)
```

Comment:
Also one observation regarding that - we do the copy only upon a `to_array` call - so this gives us the right type, but this does _not_ solve the mutability issue - if the underlying polyglot array is mutable, the calls to `wrapped_array.at` will not be referentially transparent as the underlying array may change in-between them.

I'm not saying that is an issue, because I assume that this PR is not aiming to solve the mutability issue, but I just wanted to ensure that we are on the same page there.

Reply:
Yeap, we are on the same page.